x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/session/EsqlSessionCCSUtils.java

```diff
@@ -326,6 +327,17 @@ public static void checkForCcsLicense(
             // check if it is a cross-cluster query
             if (groupedIndices.size() > 1 || groupedIndices.containsKey(RemoteClusterService.LOCAL_CLUSTER_GROUP_KEY) == false) {
                 if (EsqlLicenseChecker.isCcsAllowed(licenseState) == false) {
+                    // initialize the cluster entries in EsqlExecutionInfo before throwing the invalid license error
+                    // so that the CCS telemetry handler can recognize that this error is a CCS-related
```

Comment:
nit: comment (that I gave you, sorry...) needs adjusting. It should either end with "is CCS-related" or "is a CCS-related search".